test/truffle/cexts/puma/puma.sh

```diff
@@ -2,9 +2,8 @@
 
 source test/truffle/common.sh.inc
 
-gem_test_pack=$(jt gem-test-pack)
+jt gem install puma:3.10.0 -V -N --backtrace
+jt gem install rack:1.6.1 -V -N --backtrace
```

Comment:
I'm not wild about these versions being hard-coded and scattered around various bash files. As far as I can tell, they're not isolated either. Running `jt ruby -S puma` is going to pick up the latest version, not necessarily 3.10.0.

Reply:
Good point, although that was the same before and it seems too much to refactor these tests further in this PR, especially since they are not run in CI yet.
We could do `jt ruby -S puma _3.10.0_` potentially (but won't pin other gems), or use a Gemfile.
I think historically we have been avoiding Bundler for the `cexts` test to keep them rather minimal, but it probably doesn't matter much these days.

Setting PATH, a Gemfile.lock and `export BUNDLE_PATH=vendor/bundle` is the general approach I used in tests already using Bundler, and that isolates well.
I think it'd make sense to convert these tests to use that too to avoid installing anything globally as that can cause side effects between tests.